### ISSUE TITLE
Support upload of HEIC/HEIF files

### ISSUE THREAD

**Issue by tiwane** [inaturalist/inaturalist#2858]
From https://forum.inaturalist.org/t/support-heic-heif-images/17549

iOS devices now record photos in HEIC/HEIF format, and some people sync iPhone photos to their computer and upload them to iNat. However, in this case that won't work, the uploader throws up an error. We should look into supporting these formats on iNat, it's possible (likely?) their use will grow.

**Comment by tkriplean** [comment]
Would appreciate this! Specifically, I often drag and drop into the web interface upload page from my Photos app. It seems to work fine usually, but if I've rotated the image (and perhaps other image manipulation) via Photos, those images fail to convert in the inat interface.

**Comment by CWWhitney** [comment]
This would be a really good feature to have. It is still not working from my machine (OS 13.2 Chrome Version 110.0.5481.100). 
<img width="441" alt="Screenshot 2023-03-17 at 17 52 23" src="https://user-images.githubusercontent.com/19190662/226059850-8ac043f2-e632-4d0b-b2fb-e8724ac5daed.png">

**Comment by rainhead** [comment]
It looks like this format is supported [only by Safari](https://caniuse.com/?search=HEIC). Do you want these transcoded to JPEG?

**Comment by rainhead** [comment]
I think I remember hearing discussion about JPEG XL and WebP as well?

**Comment by geoffreywperkins** [comment]
Just adding a +1 to this feature request. I take photos on my iPhone to get location data for my observations, but for some interesting fungi I use my DSLR camera. I then put the photos from my DSLR camera onto my iPhone, then upload them to google photos, and then I focus stack the macro photos on my computer. The photos I take on the camera are RAW, which the iNaturalist iPhone app doesn't agree with, so I have to upload them via computer, which is where I run into the HEIC problem. I'm gonna keep looking for a better process in the meantime, but supporting HEIC file uploads would be super great. If BMP files could also be uploaded, that'd be great! I don't really care what format they're converted to, JPEG or JPEG XL would be fine by me!

TLDR: This would be helpful to me, converting the HEIC photos to JPEG or JPEG XL sounds great. If this could be extended to other formats like BMP, that'd be great too :)

**Comment by rainhead** [comment]
I reviewed the forum thread that lead to this issue being made. I don't see why iNaturalist should increase its maintenance burden in order to save users from having to manually export their images. Users have a wide range of workflows with a range of file types (raw or not). The iNaturalist apps do leverage their platforms to export as JPEG before upload.

I had thought before that iNaturalist was already doing image conversion somewhere in the web stack; this appears not to be true.

**Comment by geoffreywperkins** [comment]
> The iNaturalist apps do leverage their platforms to export as JPEG before upload.

Do you mean this is the case for formats like RAW? When I upload RAW photos from my iPhone, I just get the spinning wheel forever when it's trying to make the observation. Maybe it just takes a long time bc of the file size, but I've waited for it for at least a minute before, and it was still trying to upload. My wifi is pretty good, so that shouldn't be the issue, but who knows.

Regardless, not implementing this for the sake of maintenance makes sense. I think I'll just write a python script for bulk converting to JPEG. Thanks for the response :)

**Comment by rainhead** [comment]
Well, spinning wheel forever is a bug. Is this Android?

**Comment by geoffreywperkins** [comment]
@rainhead no, I use the iOS app. Also, the files that I was trying to upload when this happened were actually .NEF and .BMP

**Comment by rainhead** [comment]
@geoffreywperkins if it persists on iNaturalist Next, maybe file a bug?
